pkg/local.mk

```diff
@@ -0,0 +1,20 @@
+#
+# This file allows specifying a local folder using PKG_SOURCE_LOCAL.
+#
+# Every clean or prepare will remove $(PKG_BUILDDIR) and copy over
+# $(PKG_SOURCE_LOCAL).  This is intended to be used during package development.
+#
+# WARNING: any local changes made to $(PKG_BUILDDIR) *will* get lost!
+
+.PHONY: prepare git-download clean
+
+git-download:
+	true
+
+prepare:
```

Comment:
I would prefer doing the copy in a second target like:

```
prepare: source-copy
```

This way, `prepare` could still be used to patch the directory in the main Makefile.

Reply:
Or even keep the name `git-download` as some Makefiles depend on it like relic. (even if its during `all` and not prepare but yeah)

Reply:
prepare is always depended on and finished before the pkg makefile's main target is executed, so I think dependencies on git-download (as relic has) should work as is.

> prepare could still be used to patch the directory in the main Makefile.

This I don't understand. which is the main Makefile?

Reply:
I mean the `pkg/NAME/Makefile`. Right now many packages do a copy of makefiles during `all` (like `jerryscript` for example) while they could be doing it in `prepare`. Having the source copy done with another target, allows using 'prepare' step to copy files.


However, you are right, we should not use `git-download`. Right now some packages have `all: git-download`, which does not make any sense as prepare is executed before and so we are just trying to do it again, so overwriting `git-download` could have the bad side effect of re-executing the copy.

Reply:
right, ```all: git-download``` should be fixed...